docs/attributes.md

```diff
@@ -0,0 +1,188 @@
+# Attributes
+
+As pointed out in [Attributes as Patterns](../design-decisions/2018-07-25
+Attributes as patterns.md), attributes are encoded as Kore patterns that only
+use application and string literals.
+
+## Axiom Pattern Attributes
+
+Kore.Step.AxiomPatterns
+
+Attributes specific to interpreting axiom patterns.
+
+### Assoc
+
+Associativity axiom
+
+```
+assoc{}()
+```
+
+Axioms with this attribute should not be used for evaluating functions.
```

Comment:
The `assoc`, `comm`, `idem`, and `unit` axiom attributes identify certain axioms as associativity, commutativity, idempotency, or unit axioms. The Haskell backend does not currently do anything with those axioms, but it eventually will, so this statement is not accurate.